File: app/src/main/java/com/cere/csplayer/glide/BlurTransformation.java
package com.cere.csplayer.glide;

import android.content.Context;
import android.graphics.Bitmap;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool;
import com.bumptech.glide.load.resource.bitmap.BitmapTransformation;
import com.cere.csplayer.until.BitmapUtils;

import java.security.MessageDigest;

/**
 * Created by CheRevir on 2019/10/27
 */
public class BlurTransformation extends BitmapTransformation {
    private static final String ID = "com.cere.csplayer.glide.BlurTransformation";
    private Context mContext;

    public BlurTransformation(Context context) {
        mContext = context;
    }

    @Override
    protected Bitmap transform(@NonNull BitmapPool pool, @NonNull Bitmap toTransform, int outWidth, int outHeight) {
        return BitmapUtils.getBitmapBlur(mContext, BitmapUtils.getScaleBitmap(toTransform, 10, 10), 1.0f);
    }

    @Override
    public void updateDiskCacheKey(@NonNull MessageDigest messageDigest) {
        messageDigest.update(ID.getBytes(CHARSET));
    }

    @Override
    public int hashCode() {
        return ID.hashCode();
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        return obj instanceof BlurTransformation;
    }
}

File: app/src/main/java/com/cere/csplayer/glide/BlurTransformation.kt
package com.cere.csplayer.glide

import android.content.Context
import android.graphics.Bitmap
import android.util.Log
import com.bumptech.glide.load.engine.bitmap_recycle.BitmapPool
import com.bumptech.glide.load.resource.bitmap.BitmapTransformation
import com.cere.csplayer.until.BitmapUtils
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch
import java.security.MessageDigest

/**
 * Created by CheRevir on 2020/9/15
 */
class BlurTransformation(private val context: Context) : BitmapTransformation() {
    override fun updateDiskCacheKey(messageDigest: MessageDigest) {
        messageDigest.update(ID.toByteArray(CHARSET))
    }

    override fun transform(
        pool: BitmapPool,
        toTransform: Bitmap,
        outWidth: Int,
        outHeight: Int
    ): Bitmap {
        return BitmapUtils.getBitmapBlur(
            context,
            BitmapUtils.getBitmapScale(toTransform, 10, 10),
            1.0f
        )
    }

    override fun hashCode(): Int {
        return ID.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        return other is BlurTransformation
    }

    companion object {
        private const val ID = "com.cere.csplayer.glide.BlurTransformation"
    }
}